packages/components/src/templates/next/components/internal/CollectionCard/CollectionCard.stories.tsx

```diff
@@ -68,6 +73,18 @@ export const Hover: Story = {
   },
 }
 
+export const ExternalLink: Story = {
+  args: generateArgs({
+    isExternalLink: true,
+    title: "This is a not-so-long title that will be truncated",
+  }),
+}
+
+// TODO: fix the external link icon showing up when the text is long
```

Comment:
didn't understand this comment, what's the current behaviour when the text is long?

Reply:
my bad, should have been clearer

| current | expected |
| - | - |
| <img width="524" alt="image" src="https://github.com/user-attachments/assets/8054c01b-f3dd-4235-a676-5186a4921246" /> | <img width="534" alt="image" src="https://github.com/user-attachments/assets/06a091c8-9940-4afc-8af2-b061147675ec" /> |

couldn't find a quick easy way to do this so leaving it as a TODO - esp. since not that common for titles to be so long

Reply:
i was hesistating on this too because it's not the most ideal, tho you might be right that it's better-than-nothing. will make the UI change, but still keep the commment to explain the desired outcome!

Reply:
Ok, cc @sehyunidaaa fyi

Reply:
realized it's not as simple to apply `line-clamp-3` to the text span as that would mean the icon is always on the next line, which isn't ideal when the title is too short and not truncated.

instead, updated it (https://github.com/opengovsg/isomer/pull/1139/commits/e1fdb2d52ec8a9ae5a86a99f400fd712e6f34c40, https://github.com/opengovsg/isomer/pull/1139/commits/4cf732c7f1f2e1519ba9e9892239fcd20f456b18, https://github.com/opengovsg/isomer/pull/1139/commits/f0e2e9d83a938560d556c81fd39b50f77fc268bf) such that
- there's a `useEffect` to check if the text is truncated before deciding the location of the icon
- and because it's a useEffect, we will need `use client` so i further split it into 2 `InternalLinkTitle` (rendered during SSG) and `externalLinkTitle` (rendered during client side)

Reply:
How do I check this on storybook? Tried on [this](https://665439911775ce21bf8b8fb4-uqoozszvjs.chromatic.com/?path=/story/next-internal-components-collectioncard--external-link&globals=viewport:sm), but still was unable to see the icon

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/AWVfQY5lNPJPEvYNU3KB/b1ca478e-0734-40cd-b7ba-d4631fe22586.png)

Reply:
https://665439911775ce21bf8b8fb4-uqoozszvjs.chromatic.com/?path=/story/next-internal-components-blog-card--external-link-long-text&globals=viewport:sm

try this?